src/core/tUnification.ml

```diff
@@ -674,6 +674,19 @@ let rec unify (uctx : unification_context) a b =
 			Unify_error l ->
 				let msg = if !i = 0 then Invalid_return_type else Invalid_function_argument(!i,List.length l1) in
 				error (cannot_unify a b :: msg :: l))
+	| TFun _ , TInst(c,tl) ->
+		begin try
+			let (_,el,_) = Meta.get Meta.FunctionalInterface c.cl_meta in
```

Comment:
A check if `c` is actually an interface?

Reply:
Not here, that would be a declaration-level check. But I don't know if we should care about this.